tasks/main.yml

```diff
@@ -31,13 +43,15 @@
     state_or_province_name: "{{ ca_state_or_province_name | default(omit) }}"
 
 - name: generate ca certificate
-  openssl_certificate:
-    path: "{{ ca_openssl_path }}/{{ ca_certificate_path }}"
-    privatekey_path: "{{ ca_openssl_path }}/{{ ca_privatekey_path }}"
-    privatekey_passphrase: "{{ ca_passphrase }}"
-    csr_path: "{{ ca_openssl_path }}/{{ ca_csr_path }}"
-    provider: selfsigned
-
+  command:
```

Comment:
If it's possible to use a module like `openssl_certificate`, please do, there can be reasons why `cmd` or `command` is required.

Reply:
I left openssl_certificate because it cannot make a root certificate unfortunately.
And it can't handle certificate extensions too.